src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,974 @@
+                    BEAST_EXPECT(metadata[sfTransactionIndex.jsonName] == 1);
+                    BEAST_EXPECT(
+                        metadata[sfTransactionResult.jsonName] == "tesSUCCESS");
+                }
+            };
+
+            Json::Value tx;
+
+            tx[jss::Account] = alice.human();
+            tx[jss::TransactionType] = jss::AccountSet;
+            tx[sfDomain] = newDomain;
+            tx[sfSigners] = Json::arrayValue;
+            {
+                Json::Value signer;
+                signer[jss::Account] = becky.human();
+                signer[jss::SigningPubKey] = strHex(becky.pk().slice());
+                Json::Value signerOuter;
+                signerOuter[sfSigner] = signer;
+                tx[sfSigners].append(signerOuter);
+            }
+
+            // test with autofill
+            testTx(env, tx, validateOutput, false);
+
+            tx[sfSigningPubKey] = "";
+            tx[sfTxnSignature] = "";
+            tx[sfSequence] = env.seq(alice);
+            // transaction requires a non-base fee
+            tx[sfFee] =
+                (env.current()->fees().base * 2).jsonClipped().asString();
+            tx[sfSigners][0u][sfSigner][jss::TxnSignature] = "";
+
+            // test without autofill
+            testTx(env, tx, validateOutput);
+
+            // TODO: check that the ledger wasn't affected
+        }
+    }
+
+    void
+    testTransactionSigningFailure()
+    {
+        testcase("Transaction with a key-related failure");
+
+        using namespace jtx;
+        Env env(*this);
+        static auto const newDomain = "123ABC";
+        Account const alice{"alice"};
+        env(regkey(env.master, alice));
+        env(fset(env.master, asfDisableMaster), sig(env.master));
+        env.close();
+
+        {
+            std::function<void(Json::Value const&, Json::Value const&)> const&
+                testSimulation =
+                    [&](Json::Value const& resp, Json::Value const& tx) {
+                        auto result = resp[jss::result];
+                        checkBasicReturnValidity(
+                            result, tx, 3, env.current()->fees().base);
+
+                        BEAST_EXPECT(
+                            result[jss::engine_result] == "tefMASTER_DISABLED");
+                        BEAST_EXPECT(result[jss::engine_result_code] == -188);
+                        BEAST_EXPECT(
+                            result[jss::engine_result_message] ==
+                            "Master key is disabled.");
+
+                        BEAST_EXPECT(
+                            !result.isMember(jss::meta) &&
+                            !result.isMember(jss::meta_blob));
+                    };
+
+            Json::Value tx;
+
+            tx[jss::Account] = env.master.human();
+            tx[jss::TransactionType] = jss::AccountSet;
+            tx[sfDomain] = newDomain;
+
+            // test with autofill
+            testTx(env, tx, testSimulation, 3);
```

Comment:
last parameter must be bool

Reply:
fixed in 0bbf589f53a13069448d709faa5cea88b6ee547f